### PR TITLE
pull in latest puppet-lint plugins

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
To ensure that we pull in the latest puppet-lint plugins, we can require their latest version. Background: It's a bit hacky in pdk to update existing gems. The suggest way is to raise the gem version in the Gemfile. Now adding a dozen gems with their version to a Gemfile would make voxpupuli-puppet-lint-plugins obsolete. I think it makes sense that we from time to time update the version requirements in voxpupuli-puppet-lint-plugins.

**Comment by ekohl** [comment]
I think setting version bounds is good, but I wonder if we should make them semantically bound (like `~> 3.0`). Then also add dependabot to this repository to automatically create update PRs once it becomes available.

**Comment by bastelfreak** [comment]
@ekohl I updated it. Can you take a look again?

**Comment by bastelfreak** [comment]
dependabot is configured.

File: voxpupuli-puppet-lint-plugins.gemspec [hunk 1]
```diff
@@ -15,26 +15,26 @@ Gem::Specification.new do |s|
...
-  s.add_runtime_dependency 'puppet-lint-unquoted_string-check'
-  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase'
-  s.add_runtime_dependency 'puppet-lint-version_comparison-check'
+  s.add_runtime_dependency 'puppet-lint-absolute_classname-check', '>= 3.0.1', '< 4.0.0'
+  s.add_runtime_dependency 'puppet-lint-anchor-check', '>= 1.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-classes_and_types_beginning_with_digits-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-file_ensure-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-leading_zero-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-legacy_facts-check', '>= 1.0.4', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-lookup_in_parameter-check', '~> 1'
```
Comment (ekohl) [inline voxpupuli-puppet-lint-plugins.gemspec:24]: `~> 1` will also allow 2.0 or 3.0.
```suggestion
  s.add_runtime_dependency 'puppet-lint-lookup_in_parameter-check', '~> 1.0'
```

File: voxpupuli-puppet-lint-plugins.gemspec [hunk 2]
```diff
@@ -15,26 +15,26 @@ Gem::Specification.new do |s|
...
-  s.add_runtime_dependency 'puppet-lint-param-types'
-  s.add_runtime_dependency 'puppet-lint-resource_reference_syntax'
-  s.add_runtime_dependency 'puppet-lint-strict_indent-check'
-  s.add_runtime_dependency 'puppet-lint-top_scope_facts-check'
-  s.add_runtime_dependency 'puppet-lint-topscope-variable-check'
-  s.add_runtime_dependency 'puppet-lint-trailing_comma-check'
-  s.add_runtime_dependency 'puppet-lint-unquoted_string-check'
-  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase'
-  s.add_runtime_dependency 'puppet-lint-version_comparison-check'
+  s.add_runtime_dependency 'puppet-lint-absolute_classname-check', '>= 3.0.1', '< 4.0.0'
```
Comment (ekohl) [inline voxpupuli-puppet-lint-plugins.gemspec:18]: Isn't this sufficient?
```suggestion
  s.add_runtime_dependency 'puppet-lint-absolute_classname-check', '~> 3.0'
```

File: voxpupuli-puppet-lint-plugins.gemspec [hunk 3]
```diff
@@ -15,26 +15,26 @@ Gem::Specification.new do |s|
...
+  s.add_runtime_dependency 'puppet-lint-anchor-check', '>= 1.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-classes_and_types_beginning_with_digits-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-file_ensure-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-leading_zero-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-legacy_facts-check', '>= 1.0.4', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-lookup_in_parameter-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-manifest_whitespace-check', '>= 0.2.6', '< 1.0.0'
```
Comment (ekohl) [inline voxpupuli-puppet-lint-plugins.gemspec:25]: Semver says that on a 0.x we should treat 0.1.z -> 0.2.0 as a major version:
```suggestion
  s.add_runtime_dependency 'puppet-lint-manifest_whitespace-check', '~> 0.2.6'
```

File: voxpupuli-puppet-lint-plugins.gemspec [hunk 4]
```diff
@@ -15,26 +15,26 @@ Gem::Specification.new do |s|
...
+  s.add_runtime_dependency 'puppet-lint-param-docs', '>= 1.7.6', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-params_empty_string-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-param-types', '>= 0.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-resource_reference_syntax', '>= 1.1.0', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-strict_indent-check', '>= 2.0.8', '< 3.0.0'
+  s.add_runtime_dependency 'puppet-lint-top_scope_facts-check', '>= 1.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-topscope-variable-check', '>= 1.1.0', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-trailing_comma-check', '>= 0.4.3', '< 1.0.0'
+  s.add_runtime_dependency 'puppet-lint-unquoted_string-check', '>= 2.1.0', '< 3.0.0'
+  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '>= 1.2.0', '< 2.0.0'
```
Comment (ekohl) [inline voxpupuli-puppet-lint-plugins.gemspec:36]:
```suggestion
  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '~> 1.2.0'
```

**Review by ekohl — COMMENTED** [review]
I'm a fan of the short notation and rely on bundler to pull in the latest patch versions. Unless you really need something from it, but AFAIK we don't.

File: voxpupuli-puppet-lint-plugins.gemspec (hunk 4 above)
Comment (bastelfreak) [reply voxpupuli-puppet-lint-plugins.gemspec:36]: but that isn't the same I think? doesn't `~> 1.2.0` only allow `1.2.X`?

Comment (bastelfreak) [reply voxpupuli-puppet-lint-plugins.gemspec:36]: https://bundler.io/guides/gemfile.html
> The specifier ~> has a special meaning, best shown by example. ~> 2.0.3 is identical to >= 2.0.3 and < 2.1. ~> 2.1 is identical to >= 2.1 and < 3.0. ~> 2.2.beta will match prerelease versions like 2.2.beta.12. ~> 0 is identical to >= 0.0 and < 1.0.

Comment (ekohl) [reply voxpupuli-puppet-lint-plugins.gemspec:36]: Oh yes, I meant
```suggestion
  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '~> 1.2'
```

**Review by ekohl — APPROVED** [review]
I'm still not sure if we should bother with patch versions as minimal versions but I won't block on it.

**Pull request merged**